### DISCUSSION TITLE
What supports multiple subscribers in the async context?

### DISCUSSION THREAD

**Discussion by ylorn** [pointfreeco/swift-composable-architecture#2977] [Q&A]
Hello, just finished watching the latest episode of [Shared State: User Defaults, Part 2](https://www.pointfree.co/episodes/ep274-shared-state-user-defaults-part-2).

I noticed the [following statements](https://www.pointfree.co/episodes/ep274-shared-state-user-defaults-part-2#t75) was mentioned in the video:

> Now often an `AsyncStream` is not the correct thing to use because it does not support multiple subscribers.

So I am wondering what do support multiple subscribers for observation in the `async` context with best performance? `AsyncChannel`?

**Comment by mbrandonw** [comment]
Hi @ylorn, I believe `AsyncChannel` supports multiple subscribers (the docs don't specify unfortunately so you may need to test to make sure). But also back pressure is baked into `AsyncChannel`, which may or may not be what you want. This means that when you have to await sending data to the channel, and it will suspend until someone consumes that data.

**Reply by markst** [reply]
AsyncChannel does not support multiple subscribers. Each event is delivered to only one consumer; multiple consumers compete for elements, so some miss values.

**Comment by filblue** [comment]
Might be of interest https://github.com/apple/swift-async-algorithms/pull/357/files
